File: lua5/src/demo/runner.lua
--
-- -*- coding: utf-8 -*-
--
-- jomiel-examples
--
-- Copyright
--  2021 Toni Gündoğdu
--
--
-- SPDX-License-Identifier: Apache-2.0
--

local function dumpConfig(opts)
  local YAML = require 'lyaml/init'
  io.write(YAML.dump({ opts }))
end

local function printZMQVersion()
  local ZMQ = require 'lzmq'
  print('ZeroMQ version ' .. table.concat(ZMQ.version(), '.'))
end

local Runner = {}

function Runner.main()
  local opts = require 'demo.options'.parse()
  if opts.print_config then
    dumpConfig(opts)
  elseif opts.version_zmq then
    printZMQVersion()
  else
    local Jomiel = require 'demo.jomiel'
    Jomiel:new(opts):inquire()
  end
end

return Runner

File: python3/src/demo/runner.py
#
# jomiel-client-demos
#
# Copyright
#  2019-2021 Toni Gündoğdu
#
#
# SPDX-License-Identifier: Apache-2.0
#
from sys import exit

from demo.options import parse_options


class Runner:
    def __init__(self):
        opts = parse_options()

        if opts.print_config:
            self.dumpConfig(opts)
        elif opts.version_zmq:
            self.printZmqVersion()
        else:
            from demo.jomiel import Jomiel
            from demo.log import log_init

            log_init()
            try:
                Jomiel(opts).inquire()
            except RuntimeError as exc:
                self.handleError(exc)

    def dumpConfig(self, opts):
        from ruamel.yaml import YAML, round_trip_dump
        from sys import stdout

        yaml = YAML(typ="safe")
        yaml.default_flow_style = False
        print("---")
        round_trip_dump(opts.__dict__, stdout)
        exit(0)

    def printZmqVersion(self):
        from zmq import zmq_version

        print(f"ZeroMQ version {zmq_version()}")
        exit(0)

    def handleError(self, msg):
        from logging import error

        error(f"error: {msg}")
        exit(1)
